modules/fscloud/variables.tf

```diff
@@ -69,74 +69,15 @@ variable "allow_is_to_cos" {
   default     = true
 }
 
-variable "zone_service_ref_list" {
-  type = object({
-    cloud-object-storage        = optional(string)
-    codeengine                  = optional(string)
-    containers-kubernetes       = optional(string)
-    databases-for-cassandra     = optional(string)
-    databases-for-elasticsearch = optional(string)
-    databases-for-enterprisedb  = optional(string)
-    databases-for-etcd          = optional(string)
-    databases-for-mongodb       = optional(string)
-    databases-for-mysql         = optional(string)
-    databases-for-postgresql    = optional(string)
-    databases-for-redis         = optional(string)
-    directlink                  = optional(string)
-    iam-groups                  = optional(string)
-    is                          = optional(string)
-    messagehub                  = optional(string)
-    messages-for-rabbitmq       = optional(string)
-    schematics                  = optional(string)
-    secrets-manager             = optional(string)
-    server-protect              = optional(string)
-    user-management             = optional(string)
-    apprapp                     = optional(string)
-    compliance                  = optional(string)
-    event-notifications         = optional(string)
-    logdna                      = optional(string)
-    logdnaat                    = optional(string)
-    cloudantnosqldb             = optional(string)
-    globalcatalog-collection    = optional(string)
-    sysdig-monitor              = optional(string)
-    sysdig-secure               = optional(string)
-    toolchain                   = optional(string)
-  })
-  default = {
-    cloud-object-storage        = null
-    codeengine                  = null
-    containers-kubernetes       = null
-    databases-for-cassandra     = null
-    databases-for-elasticsearch = null
-    databases-for-enterprisedb  = null
-    databases-for-etcd          = null
-    databases-for-mongodb       = null
-    databases-for-mysql         = null
-    databases-for-postgresql    = null
-    databases-for-redis         = null
-    directlink                  = null
-    iam-groups                  = null
-    is                          = null
-    messagehub                  = null
-    messages-for-rabbitmq       = null
-    schematics                  = null
-    secrets-manager             = null
-    server-protect              = null
-    user-management             = null
-    apprapp                     = null
-    compliance                  = null
-    event-notifications         = null
-    logdna                      = null
-    logdnaat                    = null
-    cloudantnosqldb             = null
-    globalcatalog-collection    = null
-    sysdig-monitor              = null
-    sysdig-secure               = null
-    toolchain                   = null
-  }
+variable "zone_service_ref_map" {
+  type = map(object({
```

Comment:
I would keep the original structure with service names as this provide both in-context documentation and static validation as opposed to using a map here.

So:
```
type = object({
   cloud-object-storage = optional(object({
        zone_name           = optional(string)
        serviceRef_location = optional(list(string))
    })
   codeengine = ...
   etc...
```

Also the name of the variable should likely remain zone_service_ref_list as the term list was not referring to the type list but rather to the english "a list of service". We could remove the term list if this is confusing, but replacing it with the term map is not going into the right direction (possibly zone_service_refs would have been a more appropriate name, but probably fine to keep

Reply:
made the changes.